calc.js

```diff
@@ -0,0 +1,32 @@
+function getFinalResult() {
+	const pairs = [90, 10, 1, 2, 3, 4, 5, 10, 1, 90, 90, 90, 10, 22];
+	let arrOfCheckedValues = []; // wrzucamy tu wartości bez dubli
+	let arrOfObjects = []; // wrzucamy tu object {id, value} bez dubli
+	let finalResult; // ostateczny wynik, liczba
+
+	for (let i = 0; i < pairs.length; i++) {
+
+		if (arrOfCheckedValues.indexOf(pairs[i]) === -1) {
+			// filtrujemy wskazaną wartość na tablicy
+			const value = pairs.filter(item => item === pairs[i]); 
+
+			// wrzucamy na nową tablicę objekt bez dubli
+			arrOfObjects.push({ id:pairs[i], value: value.length }); 
+
+			// wrzucamy wartość którą już sprawdziliśmy przefiltrowaliśmy
+			arrOfCheckedValues.push(pairs[i]); 
+		}
+	}
+
+	// sumujemy liczby powtórzeń
+    finalResult = arrOfObjects.reduce((acc, obj) => {
+        return acc + obj.value
+    }, 0)
+
+	// zwracamy końcowy wynik
+	return finalResult;
+}
+
+// przypisujemy wartość do stałej
+const answer = getFinalResult();
+console.log(answer); // logujemy wynik - powinno wyjść 14
```

Comment:
Generalnie takie zadania wykonuje się z użyciem tzw. chaining czyli po kolei modyfikujemy pierwotną tablicę z użyciem Higher order functions, więc dobrze że użyłeś reduce, ale jeszcze by się przydała metoda map, zobacz to podejście:

const pairs = [90,10,1,2,3,4,5,10,1,90,90,90,10,22];
// podliczenie
const result = pairs
.filter((element,index,array2) => array2.indexOf(element) === index)
.map(number => ({
  [number]: pairs.filter(item => item === number).length
}))

// zsumowanie
const result2 = pairs
.filter((element,index,array2) => array2.indexOf(element) === index)
.map(number => ( pairs.filter(item => item === number).length))
.reduce((a,b) => a + b);

console.log(result);
console.log(result2);